examples/fit_ODNP_data.py

```diff
@@ -0,0 +1,171 @@
+)  
+T100_info = dict(
+    filename="T10_DI_water_230412",
+    data_dir="AG_processed_data",
+)
+# }}}
+
+# {{{ load data
+integral_vs_p = find_file(
+    data_info["filename"],
+    exp_type=data_info["data_dir"],
+    expno=f"{data_info['nodename']}/Ep",
+)
+# {{{ Some older h5 files save the T1p rather than the R1p. If there isn't an
+# R1p expno then it will load the T1p integrals and convert to R1p by taking
+# the inverse
+try:
+    R1p = find_file(
+        data_info["filename"],
+        exp_type=data_info["data_dir"],
+        expno=f"{data_info['nodename']}/R1p",
+    )
+except:
+    T1p = find_file(
+        data_info["filename"],
+        exp_type=data_info["data_dir"],
+        expno=f"{data_info['nodename']}/T1p",
+    )
+    R1p = 1 / T1p
+# }}}
+# }}}
+# {{{ The powers go up, and then go back down in order to check for
+# reproducibility. Figure out where this flip occurs
+flip_idx = np.where(np.diff(integral_vs_p.fromaxis("power").data) < 0)[0][0] + 1
+# }}}
+with figlist_var() as fl:
+    # {{{Plot integrals as a function of power
+    fl.next("Integrals vs power")
+    fl.plot(
+        integral_vs_p["power", :flip_idx],
+        "o",
+        label="Progressive Saturation Data",
+        capsize=6,
+        alpha=0.5,
+    )
+    fl.plot(
+        integral_vs_p["power", flip_idx:],
+        "rx",
+        label="Returning Power Check",
+        capsize=6,
+        alpha=0.5,
+    )
+    # }}}
+    # {{{ plot R1p data
+    fl.next(r"$R_{1}(p)$")
+    fl.plot(R1p, "o", label="Experimental Data")
+    # }}}
+    # {{{ fit kᵨ⁻¹ with two degrees of freedom (to a straight line) and then
+    # apply to fit R1p
+    T10_p = loadmat(
+        search_filename(
+            T100_info["filename"], exp_type=T100_info["data_dir"], unique=True
+        )
+    )["a"][0, :]
+    R10_p = 1 / (R1p.fromaxis("power").eval_poly(T10_p, "power"))
+    powers_fine = nddata(r_[0 : R1p.getaxis("power")[-1] : 300j], "p")
+    krho_inv = integral_vs_p.get_prop("acq_params")["concentration"] / (R1p - R10_p)
+    krho_inv_coeff = krho_inv.polyfit("power", order=1)
+    M0, A, phalf, p = symbols("M0 A phalf power", real=True)
+    R1p_expression = (T10_p[0] + T10_p[1] * p) ** -1 + (
+        integral_vs_p.get_prop("acq_params")["concentration"]
+        / (krho_inv_coeff[0] + krho_inv_coeff[1] * p)
+    )
+    R1p_fit = lambdify(p, R1p_expression)
+    fl.plot(
+        R1p_fit(powers_fine),
+        color="k",
+        label="Fit",
+        alpha=0.5,
+    )
+    plt.ylabel(r"$R_{1} / s^{-1}$")
+    plt.xlabel("Power / W")
+    # }}}
+    # {{{ Fit NMR integrals as function of power
+    fl.next("Integrals vs power")
+    sp = p / (p + phalf)
+    integral_vs_p_fit = lmfitdata(integral_vs_p["power", :flip_idx])
+    # Symbolic expression for integrals as a function of power that is used
+    # in the symbolic function for the fitting of the integrals as a function of power
+    integral_vs_p_fit.functional_form = M0 - ((M0 * A * sp) / R1p_expression)
```

Comment:
call `sp` `sp_expression`  (or use `_expression` instead of `_expr` across the board)

this is for legibility --  so you know what type of object it is